src/Class/Service/ObjectBrick/LayoutDefinitionService.php

```diff
@@ -0,0 +1,96 @@
+ */
+
+namespace Pimcore\Bundle\StudioBackendBundle\Class\Service\ObjectBrick;
+
+use Exception;
+use Pimcore\Bundle\StaticResolverBundle\Models\DataObject\ClassDefinitionResolverInterface;
+use Pimcore\Bundle\StaticResolverBundle\Models\DataObject\DataObjectResolverInterface;
+use Pimcore\Bundle\StaticResolverBundle\Models\DataObject\Objectbrick\DefinitionResolverInterface as ObjectBrickDefinitionResolverInterface;
+use Pimcore\Bundle\StudioBackendBundle\Class\Event\ObjectBrick\LayoutDefinitionEvent;
+use Pimcore\Bundle\StudioBackendBundle\Class\Hydrator\ObjectBrick\LayoutDefinitionHydratorInterface;
+use Pimcore\Bundle\StudioBackendBundle\Class\Schema\ObjectBrick\LayoutDefinition;
+use Pimcore\Bundle\StudioBackendBundle\Exception\Api\NotFoundException;
+use Pimcore\Model\DataObject;
+use Pimcore\Model\DataObject\ClassDefinitionInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * @internal
+ */
+final class LayoutDefinitionService implements LayoutDefinitionServiceInterface
+{
+    public function __construct(
+        private readonly DataObjectResolverInterface $dataObjectResolver,
+        private readonly ClassDefinitionResolverInterface $classDefinitionResolver,
+        private readonly ObjectBrickDefinitionResolverInterface $definitionResolver,
+        private readonly LayoutDefinitionHydratorInterface $layoutDefinitionHydrator,
+        private readonly EventDispatcherInterface $eventDispatcher,
+    ) {
+    }
+
+    private array $objectBrickTypes = [];
+
+    /**
+     * @inheritDoc
+     */
+    public function getLayoutDefinitionsForObject(int $dataObjectId): array
+    {
+        $dataObject = $this->dataObjectResolver->getById($dataObjectId);
+
+        $classDef = $this->classDefinitionResolver->getById($dataObject->getClassId());
+
+        $this->collectFieldCollectionTypes($classDef);
+
+        $layoutDefinitions = [];
+        foreach ($this->objectBrickTypes as $type) {
+            $layoutDefinitions[] = $this->getLayoutDefinitionByType($type);
+        }
+
+        return $layoutDefinitions;
+
+    }
+
+    /**
+     * @throws Exception
+     */
+    private function getLayoutDefinitionByType(string $name): LayoutDefinition
+    {
+        $definition = $this->definitionResolver->getByKey($name);
+
+        if (!$definition) {
+            throw new NotFoundException('Object Brick Definition', $name);
+        }
+
+        $layoutDefinition = $this->layoutDefinitionHydrator->hydrate($definition);
+
+        $this->eventDispatcher->dispatch(
+            new LayoutDefinitionEvent($layoutDefinition),
+            LayoutDefinitionEvent::EVENT_NAME
+        );
+
+        return $layoutDefinition;
+    }
+
+    private function collectFieldCollectionTypes(ClassDefinitionInterface $classDefinition): void
+    {
+        foreach ($classDefinition->getFieldDefinitions() as $fieldDefinition) {
+            if ($fieldDefinition instanceof DataObject\ClassDefinition\Data\Objectbricks) {
```

Comment:
Can we import this as whole or is there a duplicate for Objectbricks?

Reply:
Fixed it, can be imported :)